source/common/stats/thread_local_store.cc

```diff
@@ -31,6 +31,8 @@ std::list<CounterSharedPtr> ThreadLocalStoreImpl::counters() const {
     for (auto counter : scope->central_cache_.counters_) {
       if (names.insert(counter.first).second) {
         ret.push_back(counter.second);
+      } else {
+        ENVOY_LOG(warn, "duplicate counter {}.{}", scope->prefix_, counter.first);
```

Comment:
Thanks for adding the warnings for the counters and gauges; that's helpful.

Reply:
I haven't been tracking this conversation, but there should not be any warning here. It is possible that scopes overlap w/ the same stats. This can happen in different cases depending on how stats are laid out with various plugins (example: 2 listeners writing to the same HTTP stat prefix). This is why duplication is supported at the base level. Please remove the warnings here and below.

Reply:
@mattklein123 is the point you are making here that the same stat data will be shared across multiple scopes, so this function can just list to the first one?  Are there some scenarios where different scopes will want distinct stat data for the same leaf name?

Some more comments would help the naive reader, unless they are somewhere else I should've looked.